plugins/playlist-backend/src/service/router.ts

```diff
@@ -0,0 +1,271 @@
+        );
+
+      if (updatePermissionDecision.result === AuthorizeResult.CONDITIONAL) {
+        const updateConditionsFilter = transformConditions(
+          updatePermissionDecision.conditions,
+        );
+        filter = filter
+          ? { allOf: [filter, updateConditionsFilter] }
+          : updateConditionsFilter;
+      }
+    }
+
+    const playlists = await dbHandler.listPlaylists(user, filter);
+    res.json(playlists);
+  });
+
+  router.post('/', async (req, res) => {
+    await evaluateRequestPermission(req, {
+      permission: permissions.playlistListCreate,
+    });
+    const playlistId = await dbHandler.createPlaylist(req.body);
+    res.status(201).json(playlistId);
+  });
+
+  router.get('/:playlistId', async (req, res) => {
+    const { user } = await evaluateRequestPermission(req, {
+      permission: permissions.playlistListRead,
+      resourceRef: req.params.playlistId,
+    });
+    const playlist = await dbHandler.getPlaylist(req.params.playlistId, user);
+    res.json(playlist);
+  });
+
+  router.put('/:playlistId', async (req, res) => {
+    await evaluateRequestPermission(req, {
+      permission: permissions.playlistListUpdate,
+      resourceRef: req.params.playlistId,
+    });
+    await dbHandler.updatePlaylist({ ...req.body, id: req.params.playlistId });
+    res.status(200).end();
+  });
+
+  router.delete('/:playlistId', async (req, res) => {
+    await evaluateRequestPermission(req, {
+      permission: permissions.playlistListDelete,
+      resourceRef: req.params.playlistId,
+    });
+    await dbHandler.deletePlaylist(req.params.playlistId);
+    res.status(200).end();
+  });
+
+  router.post('/:playlistId/entities', async (req, res) => {
+    await evaluateRequestPermission(req, {
+      permission: permissions.playlistListUpdate,
+      resourceRef: req.params.playlistId,
+    });
+    await dbHandler.addPlaylistEntities(req.params.playlistId, req.body);
+    res.status(200).end();
+  });
+
+  router.get('/:playlistId/entities', async (req, res) => {
+    await evaluateRequestPermission(req, {
+      permission: permissions.playlistListRead,
+      resourceRef: req.params.playlistId,
+    });
+
+    const entityRefs = await dbHandler.getPlaylistEntities(
+      req.params.playlistId,
+    );
+    if (!entityRefs.length) {
+      res.json([]);
+      return;
+    }
+
+    const filter = entityRefs.map(ref => {
+      const compoundRef = parseEntityRef(ref);
+      return {
+        kind: compoundRef.kind,
+        'metadata.namespace': compoundRef.namespace,
+        'metadata.name': compoundRef.name,
+      };
+    });
+
+    const token = getBearerTokenFromAuthorizationHeader(
+      req.header('authorization'),
+    );
+
+    // TODO(kuanpg): entities in this playlist that no longer exist in the catalog will be
+    // excluded from this response, we need a way to clean up these orphaned refs potentially
+    // via catalog events (https://github.com/backstage/backstage/issues/8219)
```

Comment:
Perhaps mention here that this enforces catalog permissions as well? feeling this could easily be removed without considers that by mistake right now